homeassistant/components/twitch/sensor.py

```diff
@@ -129,9 +136,16 @@ def __init__(self, channel: TwitchUser, client: Twitch) -> None:
 
     async def async_update(self) -> None:
         """Update device state."""
-        followers = (await self._client.get_channel_followers(self._channel.id)).total
+        await self._session.async_ensure_token_valid()
+        await self._client.set_user_authentication(
+            self._session.token["access_token"],
+            OAUTH_SCOPES,
+            self._session.token["refresh_token"],
+        )
```

Comment:
Hmm, I just checked the lib, and apparently when calling this the lib does another call to check if the token is valid. Not sure if I fully like the extra request. Any thought on that?

Reply:
I don't like it either, as it's a workaround for the fact that the current version of the lib does not raise an exception when the token is expired. It is a confirmed bug - https://github.com/Teekeks/pyTwitchAPI/issues/293#issuecomment-1986418273.

I did not feel comfortable to use the `AttributeError` exception, especially sine it is thrown in an unrelated part of the code.

When that problem is fixed the correct solution would be retry logic with the refresh.

Got a better idea?

Reply:
Alternative would be this?

```
        existing_token = self._session.token["access_token"]
        await self._session.async_ensure_token_valid()
        token = self._session.token["access_token"]
        if existing_token != token:
            await self._client.set_user_authentication(
                self._session.token["access_token"],
                OAUTH_SCOPES,
                self._session.token["refresh_token"],
            )

```

Reply:
I just checked the code for `set_user_authentication`, and the solution was simple. I'll just add a False parameter and it will not make any calls. 
```
        await self._client.set_user_authentication(
            self._session.token["access_token"],
            OAUTH_SCOPES,
            self._session.token["refresh_token"],
            False,
        )
```